protocol/src/protocol.resourceChanges.proposed.md

```diff
@@ -0,0 +1,82 @@
+# Resource Changes
+
+_Client Capabilities_:
+
+The client sets the following capability if it is supporting resource Changes.
+
+```ts
+/**
+ * The workspace client capabilities
+ */
+workspace: {
+  workspaceEdit?: {
+		  /**
+		  * The client supports resource changes in `WorkspaceEdit`s
+		  */
+		  resourceChanges?: boolean;
+	  };
+}
+```
+
+_WorkspaceEdit_
+
+The following additions are proposed for __WorkspaceEdit__.
+
+
+_resourceChanges_
+
+A workspace edit can optionally provide resource changes. If present resource changes are applied before the `changes/documentChanges` has been applied
+allowing the `change/documentChanges` to work on affected files. The changes are applied in the order that they are supplied.
+
+```ts
+interface  WorkspaceEdit{
+
+  resourceChanges?: ResourceChange[]
```

Comment:
Since we talk about `changes` and `documentChanges` in a `WorkspaceEdit`, would it be better to also list those properties again here? I'm not sure what would be best since we're extending the interface here (by adding optional properties) without actually creating a new interface. Thoughts, @dbaeumer?